SwanPortAllocator/swanportallocator/portallocator.py

```diff
@@ -62,35 +63,46 @@ def __init__(self, log):
     def get_ports(self, process, n):
         """
             Return 'n' available ports and assign them to 'process' pid.
-            If the 'process' already asked for ports, and the information was not recycled, return the same ports.
             Raises NoPortsException if all ports are in use.
         """
 
         if process in self.clients:
-            ports = self.clients[process]['ports']
-
-            if len(ports) is n:
-                self.log.info('Returning same ports for process %s: %s' % (process, ports))
-                return ports
-            else:
-                n = n - len(ports)
+            stored_ports = self.clients[process]['ports']
         else:
-            ports = []
+            stored_ports = []
 
         if len(self.ports_available) < n:
```

Comment:
Since we no longer change `n`, I would put this check as the first thing we do.

Reply:
Good point, will do.